src/LinuxTracing/PerfEventReaders.cpp

```diff
@@ -162,12 +162,15 @@ struct PerfRecordSample {
   if (flags.sample_type & PERF_SAMPLE_STACK_USER) {
     ring_buffer->ReadRawAtOffset(&event.stack_size, current_offset, sizeof(uint64_t));
     current_offset += sizeof(uint64_t);
-    event.stack_data = make_unique_for_overwrite<uint8_t[]>(event.stack_size);
+    if (event.stack_size != 0u) {
+      ring_buffer->ReadRawAtOffset(
+          &event.dyn_size, current_offset + (event.stack_size * sizeof(uint8_t)), sizeof(uint64_t));
+    }
+    event.stack_data = make_unique_for_overwrite<uint8_t[]>(event.dyn_size);
     ring_buffer->ReadRawAtOffset(event.stack_data.get(), current_offset,
-                                 event.stack_size * sizeof(uint8_t));
+                                 event.dyn_size * sizeof(uint8_t));
     current_offset += event.stack_size * sizeof(uint8_t);
     if (event.stack_size != 0u) {
-      ring_buffer->ReadRawAtOffset(&event.dyn_size, current_offset, sizeof(uint64_t));
```

Comment:
Add a comment here that we move the current offset because of the dyn_size that we read before.

Reply:
Should be done!